### PR TITLE
Full support for Brazilian Portuguese

### PR DESCRIPTION
**Pull request by azolotov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ coverage
 pkg
 Gemfile.lock
 .bundle
+.idea
```
Comment (jlduran) [inline .gitignore]: Let's not commit this file

File: Gemfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ gem 'activesupport'
 group :development do
   gem 'rake'
   gem 'jeweler'
+  gem 'pry'
```
Comment (jlduran) [inline Gemfile]: Let's not commit this file

File: lib/numbers_and_words/i18n/plurals/pt.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+        extend self
+
+        def one_conditions n
+          n == 1
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/i18n/plurals/pt.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/array_joiner/languages/pt.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module NumbersAndWords
+  module Strategies
+    module ArrayJoiner
+      module Languages
+        class Pt < Base
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/array_joiner/languages/pt.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/figures_converter/decorators/pt/base.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    module FiguresConverter
+      module Decorators
+        module Pt
+          class Base < Decorators::Base
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/decorators/pt/base.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/figures_converter/decorators/pt/fractional.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+            def fraction_length
+              @options[:fractional][:length].to_i
+            end
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/decorators/pt/fractional.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/figures_converter/decorators/pt/integral.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+            def figures
+              @strategy.language.figures
+            end
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/decorators/pt/integral.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/figures_converter/options/pt/gender.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+            def active?
+              @options[:gender]
+            end
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/options/pt/gender.rb]: Missing line at the end

File: lib/numbers_and_words/strategies/figures_converter/options/pt/remove_zero.rb [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+            def active?
+              @options[:remove_zero]
+            end
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jlduran) [inline lib/numbers_and_words/strategies/figures_converter/options/pt/remove_zero.rb]: Missing line at the end

File: spec/numbers_and_words/float/fixture_examples/pt.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    0.0001: um décimo de milésimo
+    0.10001: dez mil e um centésimos de milésimos
+    0.100001: cem mil e um milionésimos
+    0.0101001: cento e um mil e um décimos de milionésimos
+    0.100000010000001: cem trilhões, dez milhões e um quadrilionésimos
+    0.000110000000030000000001: onze bilhões e três centésimos de trilionésimo
+    1.0: um
+    0.7: sete décimos
+    3.5: três e cinco décimos
+    42.1: quarenta e dois e um décimo
```
Comment (jlduran) [inline spec/numbers_and_words/float/fixture_examples/pt.yml]: Missing line at the end

**Comment by jlduran** [comment]
Thank you!

I have only one more question:

Does Portugal use the long scale of numbers? I know Brazil uses the short scale. I am not familiar with the _Acordo Ortográfico da Língua Portuguesa de 1990_.

If so, this should be changed to `pt-BR`.

**Comment by azolotov** [comment]
@jlduran We realized that, too, and are already working on switching it to pt-BR, so the original pt implementation will remain and we're adding an entirely new pt-BR one with this pull request. cc @rafaelcaricio

**Comment by rafaelcaricio** [comment]
I will update the PR soon enough. 😃

**Comment by rafaelcaricio** [comment]
@jlduran @azolotov should be good to go now.

File: spec/numbers_and_words/integer/fixture_examples/pt-BR.yml [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    2017: dois mil e dezessete
+    2045: dois mil e quarenta e cinco
+    4000: quatro mil
+    5000: cinco mil
+    5002: cinco mil e dois
+    11000: onze mil
+    21000: vinte e um mil
+    21001: vinte e um mil e um
+    999000: novecentos e noventa e nove mil
+    999999: novecentos e noventa e nove mil e novecentos e noventa e nove
```
Comment (jlduran) [inline spec/numbers_and_words/integer/fixture_examples/pt-BR.yml]: Could you verify this is correct? I think it should be how it actually is (if so, other numbers should change as well):

```yml
999999: novecentos e noventa e nove mil novecentos e noventa e nove
```

Looking at a few documents:

https://www.legisweb.com.br/legislacao/?id=178282 Art. 3º, Seção 2º

> ... o número nove milhões, novecentos e noventa e nove mil novecentos e noventa e nove ...

Also:
 
http://www.languagesandnumbers.com/how-to-count-in-portuguese-brazil/en/por-bra/

Comment (rafaelcaricio) [inline spec/numbers_and_words/integer/fixture_examples/pt-BR.yml]: You are right. Just keep in mind that the website "languagesandnumbers.com" does not add the required commas, but we are adding in our implementation. So we are correct in that matter and the site wrong.

**Comment by jlduran** [comment]
Thank you. I think this is good to go now. I will merge it right now, however I'm not sure if I can cut a new gem today.

Let's ask @dblock, otherwise by the end of the week I'll have it ready.

**Pull request merged**